fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -66,11 +69,11 @@ def __init__(
         X,
         y,
         estimator,
-        constraints,
-        B,
-        objective=None,
-        opt_lambda=True,
-        sample_weight_name="sample_weight",
+        constraints: Moment,
+        B: float,
+        objective: Moment | None = None,
```

Comment:
theoretically, this could be `Optional[Moment] = None` if you want to import `Optional` from `typing`. Just a suggestion...

Reply:
I think I prefer to stick with the new python syntax :sweat_smile: no strong preference though